jwst/tests_nightly/general/nirspec/test_nirspec_steps_single.py

```diff
@@ -345,3 +346,49 @@ def test_nirspec_mos_masterbg_user(self):
         outputs = [(result_file, ref_file)]
         self.compare_outputs(outputs)
         result.close()
+
+@pytest.mark.bigdata
+class TestNIRSpecMasterBackgroundNodded(BaseJWSTTest):
+    input_loc = 'nirspec'
+    ref_loc = ['test_masterbackground', 'nrs-ifu', 'nodded', 'truth']
+    test_dir = ['test_masterbackground', 'nrs-ifu', 'nodded']
+
+    rtol = 0.000001
+
+    def test_nirspec_masterbg_nodded(self):
+        """Run masterbackground step on NIRSpec association"""
+        asn_file = self.get_data(*self.test_dir,
+                                  'nirspec_spec3_asn.json')
+        for file in raw_from_asn(asn_file):
+            self.get_data(*self.test_dir, file)
+
+        collect_pipeline_cfgs('./config')
+        result = MasterBackgroundStep.call(
+            asn_file,
+            config_file='config/master_background.cfg',
+            save_background=True,
+            save_results=True
+            )
+
+        # test 1
+        # compare  background subtracted data  to truth files
+        # check that the  cal_step master_background ran to complete
+        for model in result:
+            assert model.meta.cal_step.master_background == 'COMPLETE'
+
+            result_file = model.meta.filename.replace('cal', 'master_background')
+            truth_file = self.get_data(*self.ref_loc, result_file)
+
+            outputs = [(result_file, truth_file)]
+            self.compare_outputs(outputs)
+
+
+        # test 2
+        # compare the master background combined file to truth file
+        master_combined_bkg_file = 'ifu_prism_source_off_fix_NRS1_o001_masterbg.fits'
+        truth_background = self.get_data(*self.ref_loc,
+                                          master_combined_bkg_file)
+        outputs = [(master_combined_bkg_file, truth_background)]
+        self.compare_outputs(outputs)
```

Comment:
`compare_outputs` can compare many outputs.  I would compare them all at once.  It should only be called once per test.  So use the loop above to generate the outputs to compare, append the computed master background file, then send them all to `compare_outputs`.

See coron pipeline for an example:

https://github.com/spacetelescope/jwst/blob/7e412b5b0138bf2fb5dca61e87ecedb205a2c58b/jwst/tests_nightly/general/nircam/test_coron3_1.py#L37-L62